zebra-chain/src/transaction.rs

```diff
@@ -136,4 +144,33 @@ impl Transaction {
         self.inputs()
             .any(|input| matches!(input, TransparentInput::Coinbase { .. }))
     }
+
+    // TODO(jlusby): refine type
+    pub fn sighash(&self, network: Network, height: BlockHeight, hash_type: u32) -> Hash {
```

Comment:
`to_sighash` maybe?

Reply:
`to_foo` suggests that the function is a conversion to `foo`, but I wouldn't really call this a conversion, so I think that `sighash` is better.

Reply:
It depends on if you consider a `sighash` an alternate representation for (self, network, height, hash_type). If so then `to_*` would be appropriate according to [the conv api guidelines](https://rust-lang.github.io/api-guidelines/naming.html#c-conv). That said, I can't find any `to_` conversions that would be analogous to hashing in `std`, the only function is `hash` on the `Hash` trait, which seems to back up Henry. /shrug

Reply:
Yeah, I personally would not consider a sighash to be an alternate representation.

Reply:
We already have Block.hash(), so let's try to be internally consistent as well.

Reply:
In another PR that got closed when I added a `.hash` function to `Transaction` @dconnolly suggested switching the name to something other than `.hash` on the basis of it potentially causing method resolution conflicts with methods on traits like `Hash` and `Digest`, so we renamed `hash` in that PR to `to_hash` because it seemed vaguely appropriate and resolved the name conflict issue, and on the basis of that PR I vaguely been planning on renaming the `hash` function on `Block`.

Reply:
> In another PR that got closed when I added a `.hash` function to `Transaction` @dconnolly suggested switching the name to something other than `.hash` on the basis of it potentially causing method resolution conflicts with methods on traits like `Hash` and `Digest`. We renamed `hash` in that PR to `to_hash` because it seemed vaguely appropriate and resolved the name conflict issue. On the basis of that PR I vaguely been planning on renaming the `hash` function on `Block`.

Yeah all of this

Reply:
I think `Block::hash` is a good name for getting the hash of a block. I don't think there's a conflict with the `Hash` methods because `Block` doesn't implement `Hash` (and shouldn't).

Reply:
Ok then I'm good with `hash()/sighhash()` 😁